05.wc/wc.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  paths, target_options = options_and_paths
+  text_metadata_collection =
+    if paths.empty?
+      [create_text_metadata(input: $stdin.read)]
+    else
+      build_text_metadata(paths)
+    end
+  width = calculate_output_width(text_metadata_collection, target_options)
+  text_metadata_collection.each do |text_metadata|
+    render(target_options, text_metadata, width)
+  end
+end
+
+def options_and_paths
+  line = false
+  word = false
+  size = false
+  opt = OptionParser.new
+  opt.on('-l') { |v| line = v }
+  opt.on('-w') { |v| word = v }
+  opt.on('-c') { |v| size = v }
+  paths = opt.parse(ARGV)
+  target_options =
+    if [line, word, size].none?
+      %i[line_count word_count size]
+    else
+      { line_count: line, word_count: word, size: size }.map do |option, flag|
+        option if flag
+      end.compact
+    end
+  [paths, target_options]
+end
+
+def create_text_metadata(input:, name: nil)
+  {
+    line_count: input.lines.count,
+    word_count: input.split.count,
+    size: input.size,
+    name: name
+  }
+end
+
+def build_text_metadata(paths)
+  text_metadata_collection = paths.map do |path|
+    create_text_metadata(input: File.read(path), name: path)
+  end
+  if text_metadata_collection.length > 1
+    text_metadata_collection + [calculate_total(text_metadata_collection)]
+  else
+    text_metadata_collection
+  end
```

Comment:
```suggestion
  text_metadata_collection << calculate_total(text_metadata_collection) if paths.size > 1
  text_metadata_collection
```

好みの問題ですが、こんなふうに書くのもありです

Reply:
提示いただいた方が簡潔で好みでした！[冗長であったため修正](https://github.com/s-tone-gs/ruby-practices/pull/10/commits/200be156b8c7b2a1c6a1bad10f69be4a971e3ad3)にて反映しました！